src/utilities/copyTextToClipboard.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copies a piece of text to the clipboard. Adapted from [StackOverflow][1].
+ *
+ * @param text the text to copy
+ * @returns whether the operation was successful
+ */
+export async function copyTextToClipboard(text: string): Promise<boolean> {
+  if (navigator.clipboard) {
+    try {
+      await navigator.clipboard.writeText(text)
+      return true
+    } catch {
+      // Do nothing and let the remaining logic try to be successful.
+    }
+  }
+
+  const textArea = document.createElement('textarea')
+
+  // Styles for ensuring browser support
+  textArea.style.position = 'fixed'
+  textArea.style.top = '0'
+  textArea.style.left = '0'
+  textArea.style.width = '32px'
+  textArea.style.height = '32px'
+  textArea.style.padding = '0'
+  textArea.style.border = 'none'
+  textArea.style.outline = 'none'
+  textArea.style.boxShadow = 'none'
+  textArea.style.background = 'transparent'
+
+  textArea.value = text
+
+  document.body.appendChild(textArea)
+  textArea.focus()
+  textArea.select()
+
+  let isSuccess: boolean
+
+  try {
+    isSuccess = document.execCommand('copy')
+  } catch {
+    isSuccess = false
+  } finally {
+    document.body.removeChild(textArea)
+  }
+
+  return isSuccess
+}
```

Comment:
In theory it would be possible, but I really don’t like the KClipboardProvider component in principle.

All it does is wrap a utility function into a component only to pass the function back via scoped slots so you can use it as the `@click` listener. That’s just unnecessary. Unfortunately Kongponents won’t provide the plain utility function so I’m copying it.

Reply:
@adamdehaven what's your take on the "us not using the kongponent thing to do copying"?

Reply:
Not arguing that the Kongponents utility couldn't be improved; however, IMO I'd use the Kongponent `KClipboardProvider` just like all of the other Kong apps. Unless I'm missing something, I don't see much enhancement by the code being implemented here, even if it extracts away the slot usage

Reply:
Ok sounds like to me then then if we don't have a good technical reason not to use it then we should. I think the other thing worth saying is that if its only being used in the CopyButton component (I can't think of another place where we use copy to clipboard functionality apart from in a CopyButton component), we will only be using it in a single place and its usage is hidden from the app, so you can pretend its not there if you don't like using it. I think this is important as even if the reason is "its awkward to use" which is some cases could be valid, we only have to use it once so it becomes a non-issue.

Let me know if you have a good strong technical reason not to use it, otherwise I would go with the Kongponent. lmk

Reply:
Fine. I made that change.

I will once more remark that KClipboardProvider is needlessly a component when all it does provide the copy function via a slot.